p2p/src/authenticated/discovery/actors/listener.rs

```diff
@@ -33,9 +67,21 @@ pub struct Actor<
 
     address: SocketAddr,
     stream_cfg: StreamConfig<C>,
-    rate_limiter: RateLimiter<NotKeyed, InMemoryState, E, NoOpMiddleware<E::Instant>>,
+    max_concurrent_handshakes: u32,
+    in_flight_handshakes: Arc<AtomicU32>,
+    ip_rate_limiter:
+        Arc<RateLimiter<IpAddr, HashMapStateStore<IpAddr>, E, NoOpMiddleware<E::Instant>>>,
+    subnet_rate_limiter:
+        Arc<RateLimiter<SubnetKey, HashMapStateStore<SubnetKey>, E, NoOpMiddleware<E::Instant>>>,
+    peer_rate_limiter: Arc<
+        RateLimiter<C::PublicKey, HashMapStateStore<C::PublicKey>, E, NoOpMiddleware<E::Instant>>,
+    >,
+    prioritized_ips: Arc<RwLock<HashMap<IpAddr, HashSet<C::PublicKey>>>>,
```

Comment:
nit: change this to known_ips (they are only prioritized in that we assume some specified peer is dialing from there)